haystack/nodes/file_converter/azure.py

```diff
@@ -151,15 +178,19 @@ def _convert_tables_and_text(
         meta: Optional[Dict[str, str]],
         valid_languages: Optional[List[str]],
         file_path: Path,
-    ) -> List[Dict[str, Any]]:
-        tables = self._convert_tables(result, meta)
-        text = self._convert_text(result, meta)
+        id_hash_keys: Optional[List[str]] = None,
+    ) -> List[Document]:
+        tables = self._convert_tables(result, meta, id_hash_keys)
+        text = self._convert_text(result, meta, id_hash_keys)
         docs = tables + [text]
 
         if valid_languages:
-            file_text = text["content"] + " ".join(
-                [cell for table in tables for row in table["content"] for cell in row]
-            )
+            file_text = text.content
+            for table in tables:
+                assert isinstance(table.content, pd.DataFrame)
```

Comment:
I'd rather raise a proper `HaystackError` in here with a description of what's wrong

Reply:
Good point! Agreed. 👍

Reply:
@ZanSara @julian-risch These assert statements are needed for the mypy checks. The `Document`'s content field can be either of type `str` or `pd.DataFrame`. With these assert statements, we tell mypy that we are certain that `table.content` is of type `pd.DataFrame`. Otherwise, we would get a type error, because elements of type `str` don't have a method `iterrows`. 

The alternative would be to use `# type: ignore`, but AFAIK we try to avoid these as much as possible.

Reply:
Good point, I changed the related line of code.